syntax = "proto3";

import "PrivateOrdersV3Api.proto";
import "PrivateDealsV3Api.proto";
import "PrivateAccountV3Api.proto";

option java_package = "com.mxc.push.common.protobuf";
option optimize_for = SPEED;
option java_multiple_files = true;
option java_outer_classname = "PushDataV3ApiWrapperProto";

message PushDataV3ApiWrapper {

  // Channel name, e.g. "spot@private.orders.v3.api.pb"
  string channel = 1;

  oneof body {
    PrivateOrdersV3Api privateOrders = 304;
    PrivateDealsV3Api privateDeals = 306;
    PrivateAccountV3Api privateAccount = 307;
  }

  // Optional common metadata
  optional string symbol = 3;
  optional string symbolId = 4;
  optional int64 createTime = 5;
  optional int64 sendTime = 6;
}


